modules/gapi/include/opencv2/gapi/streaming/onevpl/accel_types.hpp

```diff
@@ -0,0 +1,72 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2022 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_ACCEL_TYPES_HPP
+#define GAPI_STREAMING_ONEVPL_ACCEL_TYPES_HPP
+
+#include <limits>
+#include <string>
+
+#include "opencv2/gapi/own/exports.hpp" // GAPI_EXPORTS
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+enum class AccelType: uint8_t {
+    HOST,
+    DX11,
+
+    LAST_VALUE = std::numeric_limits<uint8_t>::max()
+};
+
+GAPI_EXPORTS const char* to_cstring(AccelType type);
```

Comment:
why not?

Reply:
I'd propose to hide it if possible, in `device_selector_interface.hpp` for example. As far as I can see, this is the implementation detail and external user shouldn't have access to this function

Reply:
it's not implementation detail: it's public field which described public Device and Context types.
Since Device/Context is public and provides public method `AccelType get_type() `  then it's good option to provide human-readable type description

Reply:
Oh, how I see

Reply:
"from classic"
also std::string  is heavy for simple const-cstring returning operation